### PR TITLE
Self docs

### PR DESCRIPTION
**Pull request by johanfylling**
docs: Adding documentation for the `self.*` built-in functions

### CODE REVIEW

**Comment by johanfylling** [comment]
Only consider changes under `docs` when reviewing. Other changes are covered by https://github.com/open-policy-agent/opa/pull/4537

**Comment by johanfylling** [comment]
I rebased and excluded the `self` impl. To make it easier to review. Please note that `self` built-ins aren't operational in this branch.

File: docs/content/annotations.md [hunk 1]
```diff
@@ -50,149 +50,35 @@ a metadata block determines how that metadata block will be applied. If the
...
-    access[_] == input.operation
-}
-```
-
-This is obviously redundant and error-prone. To avoid this problem, we can
-define the metadata block once on a rule with scope `document`:
+#### Example
 
-```
+```rego
```
Comment (srenatus) [inline docs/content/annotations.md:67]: I think this should be a live-block (maybe a read_only one, but still), for consistent formatting and handling of Rego snippets on the website.

File: docs/content/annotations.md [hunk 2]
```diff
@@ -407,7 +293,7 @@ compiler.Compile(modules)
 as := compiler.GetAnnotationSet()
 ```
 
-The `ast.AnnotationSet` can be flattened into a slice of `ast.AnnotationsRef`, which is a complete, sorted list of all 
+The ``ast.AnnotationSet`` can be flattened into a slice of ``ast.AnnotationsRef``, which is a complete, sorted list of all
```
Comment (srenatus) [inline docs/content/annotations.md:296]: I'm confused here. What's the meaning of one backtick vs two backticks?

**Review by srenatus — COMMENTED** [review]
Some markup-related remarks only 👍

File: docs/content/annotations.md (hunk 1 above)
Comment (johanfylling) [reply docs/content/annotations.md:67]: I'll look into that 👍 .

**Comment by johanfylling** [comment]
I've pulled these changes into https://github.com/open-policy-agent/opa/pull/4537

**Pull request closed without merging**